daemon/README

```diff
@@ -0,0 +1,193 @@
+Existing User Tracking
+======================
+
+- tracking is done by:
+  - monitoring /etc/passwd file
+
+- maintained data:
+  - minimum UID
+  - maximum UID
+  - set of valid UID values
+
+- UID added:
+  - no action needed
+
+- UID removed:
+  - user must be dropped from settings data [NOT DONE]
+
+User Session Tracking
+=====================
+
+- tracks UID of current user
+
+- UID is used for:
+  - permissions checking/prompting
+
+- UID changed:
+  - cancels pending/queued prompts [NOT DONE]
+
+Permission File Tracking
+========================
+
+- tracking is done by:
+  - monitoring /etc/sailjail/permissions directory
+
+- maintained data:
+  - set of PERMISSION names
+
+- PERMISSION added/removed:
+  - triggers application data re-evaluation
+
+Desktop File Tracking
+=====================
+
+- tracking is done by:
+  - monitoring /usr/share/applications directory
+
+- maintained data:
+  - set of APPLICATION data, each with:
+    - Desktop properties: Name, Type, Exec, Icon
+    - Sailjail properties: Permissions
+
+- APPLICATION added:
```

Comment:
Any sensible add/remove/edit permissions/users/applications/etc needs to lead into daemon performing appropriate actions whether it is running or stopped at the time of change.

Applications are relevant only in the context of user settings. When application is added -> implicit defaults are used. When application is removed, user settings related to that application need to be flushed. When application is edited, user settings are re-evaluated.

Reply:
So, from data state point of view it already should not matter whether applications get installed / upgraded / removed during daemon runtime / downtime. 

But from change signaling point of view there is the question of: do we want to emit change signals on startup (currently this is not done) / should client interested in that level of detail re-enumerate when service appears on system bus. If we want to allow "enumerate once and then only track changes" like client behavior, then yes, also the "previously signaled applications" list needs to be made persistent.

Reply:
My question was mostly about, will the application installation be correct at the OS update stage, provided that the permission daemon is not running? Or should we start the daemon earlier?

From the data monitoring point of view, I guess "enumerate once + subscribe" could work.

Reply:
The idea is that daemon does not need to be running during update stage. It should already work. If it does not, it is a bug.

Personally I prefer "subscribe + enumerate on name owner change" client side behavior which more or less makes service startup time signaling irrelevant without adding complexity vs "enumerate once + subscribe". If we want to support also the latter more fragile approach, it can be done - once we get the initial version in.